### ISSUE TITLE
[Bug] Cannot fit GP with ConstantMean module 

### ISSUE THREAD

**Issue by mardebug** [cornellius-gp/gpytorch#1139]
# 🐛 Bug

I'm trying to fit 3 data points with a GP with the  `ConstantMean()` mean module. However, I can only fit the 3 points if I'm using the `ZeroMean()` and if I don't normalize the labels (`with_constant_mean= False` and `with_normalized_labels = False`). If I change to any other combination of the two (e.g. using `ConstantMean()` and/or non-normalized y), the points are not fitted at all.

It doesn't matter whether I use double or single precision (`with_double`).

## To reproduce

I adopted the code from [here](https://github.com/cornellius-gp/gpytorch/issues/728#issuecomment-499550305)

** Code snippet to reproduce **
```python
import math
import torch
import gpytorch
import numpy as np
from matplotlib import pyplot as plt

with_double = False
with_normalized_labels = False
with_constant_mean = True

bounds = np.array([[0,1]])
train_x = torch.linspace(bounds[0,0], bounds[0,1], 3)
if with_double:
    train_x = train_x.double()
train_y = np.sin(60 * train_x ** 4)
if with_normalized_labels:
    train_y -= train_y.mean()

# We will use the simplest form of GP model, exact inference
class ExactGPModel(gpytorch.models.ExactGP):
    def __init__(self, train_x, train_y, likelihood):
        super(ExactGPModel, self).__init__(train_x, train_y, likelihood)
        if with_constant_mean:
            self.mean_module = gpytorch.means.ConstantMean()
        else:
            self.mean_module = gpytorch.means.ZeroMean()
        self.covar_module = gpytorch.kernels.ScaleKernel(gpytorch.kernels.MaternKernel())

    def forward(self, x):
        mean_x = self.mean_module(x)
        covar_x = self.covar_module(x)
        return gpytorch.distributions.MultivariateNormal(mean_x, covar_x)

# initialize likelihood and model
likelihood = gpytorch.likelihoods.GaussianLikelihood()
model = ExactGPModel(train_x, train_y, likelihood)
if with_double:
    model.double()

print("Initialized with: lengthscale=%.3f variance=%.3f noise=%.5f" % (model.covar_module.base_kernel.lengthscale.item(),
        model.covar_module.outputscale.item(),
        model.likelihood.noise.item()))

# Find optimal model hyperparameters
model.train()
likelihood.train()

# Use the adam optimizer
optimizer = torch.optim.Adam([
    {'params': model.parameters()},
], lr=0.1)

# "Loss" for GPs - the marginal log likelihood
mll = gpytorch.mlls.ExactMarginalLogLikelihood(likelihood, model)

training_iter = 10000
for i in range(training_iter):
    # Zero gradients from previous iteration
    optimizer.zero_grad()
    # Output from model
    output = model(train_x)
    # Calc loss and backprop gradients
    loss = -mll(output, train_y)
    loss.backward()
    print('Iter %d/%d - Loss: %.3f   lengthscale: %.3f, variance: %.3f,   noise: %.5f' % (
        i + 1, training_iter, loss.item(),
        model.covar_module.base_kernel.lengthscale.item(),
        model.covar_module.outputscale.item(),
        model.likelihood.noise.item()
    ))
    optimizer.step()

# Prediction
# Get into evaluation (predictive posterior) mode
model.eval()
likelihood.eval()

# Test points are regularly spaced along [0,1]
# Make predictions by feeding model through likelihood
with torch.no_grad(), gpytorch.settings.fast_pred_var():
    test_x = torch.linspace(bounds[0,0], bounds[0,1], 1000)
    if with_double:
        test_x = test_x.double()
    observed_pred = likelihood(model(test_x))
    observed_pred_mean = observed_pred.mean

    # Initialize plot
    f, ax = plt.subplots(1, 1, figsize=(10, 10))

    # Get upper and lower confidence bounds
    lower, upper = observed_pred.confidence_region()
    # Plot training data as black stars
    ax.plot(train_x.numpy(), train_y.numpy(), 'k*')
    # Plot predictive means as blue line
    ax.plot(test_x.numpy(), observed_pred_mean.numpy(), 'b')
    # Shade between the lower and upper confidence bounds
    ax.fill_between(test_x.numpy(), lower.numpy(), upper.numpy(), alpha=0.5)
    ax.set_ylim([-3, 3])
    ax.legend(['Observed Data', 'Mean', 'Confidence'])

```

Generated with setting 
`with_normalized_labels = False` and `with_constant_mean = True`
or by setting
`with_normalized_labels = True` and `with_constant_mean = False`,
this code yields:

![labels_F_mean_T](https://user-images.githubusercontent.com/65032133/81426523-9a982300-9159-11ea-8a9f-69b6ca735b03.PNG)

## Expected Behavior

I would expect that `with_constant_mean = True` yields the same result as below.

Expected behavior can be generated by setting `with_constant_mean= False` and `with_normalized_labels = False`.

![expected_behavior](https://user-images.githubusercontent.com/65032133/81426515-953ad880-9159-11ea-9794-13f90ef43d63.PNG)

## System information

**Please complete the following information:**
- GPyTorch Version: 1.1.1
- PyTorch Version: 1.5.0+cu101
-  Computer OS: Ubuntu 18.04

**Comment by gpleiss** [comment]
This isn't a bug - but simply the reality when you are trying to learn a model with only 3 data points :)

There are several different hyperparameter settings that could possibly fit this data (e.g. all data points are essentially noise, the data points actually represent some pattern, etc.). The constant mean adds a bit of "expressive power" to the model (one additional learnable parameter), which might result in the optimization getting stuck in the local minimum that corresponds to all points being noise.

Again, with only 3 data points, both model could arguably be correct. Note that - if you modify the hyperparameter initializations slightly (e.g. `model.likelihood.initialize(noise=0.01)` before the training loop) you get a different fit.

WIth such a small dataset, optimizing the hyperparameters is likely to cause overfitting. I'd suggest doing full Bayesian inference (i.e. sampling over the hyperparameters) as in this tutorial: https://gpytorch.readthedocs.io/en/latest/examples/01_Exact_GPs/GP_Regression_Fully_Bayesian.html
